### PR TITLE
Don't cache the secret, extra logs

### PR DESCRIPTION
**Pull request by costinm**
Note that once node-agent is deprecated ( and I think in master it should be ), we can
remove the cache completely and simplify the code. 

The cache was needed when multiple pods in a node where sharing an agent.

### CODE REVIEW

**Comment by howardjohn** [comment]
Don't we still need nodeagent for 1.5 to support migration from 1.4 SDS -> 1.5, or SDS upgrade is already broken so it doesn'ty matter if we remove the nodeagent?

**Comment by howardjohn** [comment]
/retest

I am tracking down the failure in https://github.com/istio/istio/issues/20487

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -214,7 +216,54 @@ func NewServer(args *PilotArgs) (*Server, error) {
...
+				}()
+				return nil
+			})
+
+		} else {
+			s.caBundlePath = path.Join(localCertDir.Get(), "cert-chain.pem")
+		}
+	}
+
+		// initDNSListener() must be called after the createCA()
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go]: nit: extra space, probably lint will fail

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -214,7 +216,54 @@ func NewServer(args *PilotArgs) (*Server, error) {
 
 	}
 
-	// initDNSListener() must be called after the createCA()
+	// Setup the root cert chain and caBundlePath - before calling initDNSListener.
+	if features.PilotCertProvider.Get() == KubernetesCAProvider {
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go]: nit: why not just have this in initDnsListener? Right now this function is mostly just calling out to a bunch of other functions so it seems out of place to have all of this logic here

Comment (ayj) [reply pilot/pkg/bootstrap/server.go]: I don't remember why I put it here. `initDnsListener()`sounds like the right place to put it.

**Pull request merged**